src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs

```diff
@@ -102,6 +103,10 @@ internal class PayloadHelper
                         SerDe.Write(destination, (double)arg);
                         break;
 
+                    case TypeCode.DateTime:
+                        SerDe.Write(destination, ((DateTime)arg).ToString("yyyy-MM-dd"));
```

Comment:
`FromInternal` is getting an integer as a datetime value. Is it possible to pass the integer as value instead of string or is this required? Do you know which code on the scala is changing this back to sql datetype?

Reply:
The code [here](https://github.com/dotnet/spark/blob/master/src/scala/microsoft-spark-2.4.x/src/main/scala/org/apache/spark/api/dotnet/SerDe.scala#L84) reads datatype as string.